client.go

```diff
@@ -2893,8 +2893,8 @@ func (t *transport) RoundTrip(hc *HostClient, req *Request, resp *Response) (ret
 
 	br := hc.acquireReader(conn)
 	err = resp.ReadLimitBody(br, hc.MaxResponseBodySize)
-	hc.releaseReader(br)
 	if err != nil {
+		hc.releaseReader(br)
```

Comment:
@erikdubbelboer this is another change that would be required to avoid data races if the connection can be closed from outside. The reader shall only be released directly if it is not a stream connection. Otherwise do it in the cleanup method called via the closer interface.

The reason is that if the reader is realeased earlier, it would then be available for the next request and might cause a data race.